src/main/java/com/google/research/bleth/simulator/BoardState.java

```diff
@@ -105,6 +108,17 @@ public String toJson() {
         return new Gson().toJson(this.matrix);
     }
 
+    /** Returns a map that maps to each populated location the representations (type and id) of the agents on this location. */
+    public Multimap<Location, String> agentsRepresentationsOnStateBoard() {
+        Multimap<Location, String> locationsToRepresentations = ArrayListMultimap.create();
```

Comment:
Consider just using a `new ImmutalbeListMultimap.Builder<Location, String>` for this and avoiding the need for a copy later.